File: 12B/Programming/pa3/DictionaryTest.java
//-----------------------------------------------------------------------------
// DictionaryTest.java
// Sinclair Liang
// wliang13@ucsc.edu
// pa3: This file is to test different functionalities in Dictionary ADT
//-----------------------------------------------------------------------------
import java.util.Scanner;
import java.io.*;

class DictionaryTest
{
   public static void main(String[] args)
   {
   	String v;
   	Dictionary A = new Dictionary();
   	A.insert("3","e");
   	A.insert("8","f");
   	A.insert("56","54");
   	// System.out.println(A.isEmpty());
   	// System.out.println(A.size());
   	// v = A.lookup("3");
      System.out.println(A);
      // A.delete("3");
      // A.delete("8");
      A.delete("56");
      // System.out.println(A.size());

   	// System.out.println(v);
   	System.out.println(A);
   	// A.makeEmpty();
   	// System.out.println(A);
   }
}

File: 12B/Programming/pa5/DictionaryTest.c
//-----------------------------------------------------------------------------
// 	DictionaryTest.c
//  Sinclair Liang
//  wliang13@ucsc.edu
// 	Test file for the Dictionary ADT with Hash Table
//-----------------------------------------------------------------------------


#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[])
{
	Dictionary A = newDictionary();
	char* k;
	char* v;
	char* word1[] = {"one", "two", "three", "four", "five", "six", "seven"};
	char* word2[] = {"foo", "bar", "blah", "galumph", "happy", "sad", "blue"};
	int i;
	insert(A, word1[0], word1[1]);
	insert(A, word1[1], word1[2]);
	// delete(A, "one");
	// delete(A, "seven");
	printDictionary(stdout, A);


	k = word1[0];
	v = lookup(A, k);
	printf("key=\"%s\" %s\"%s\"\n", k, (v == NULL ? "not found " : "value="), v);
	printf("%s\n", (isEmpty(A) ? "true" : "false"));
	makeEmpty(A);
	printf("%s\n", (isEmpty(A) ? "true" : "false"));
	freeDictionary(&A);

}
